tests/components/xiaomi_ble/test_device_trigger.py

```diff
@@ -0,0 +1,246 @@
+    inject_bluetooth_service_info_bleak(
+        hass,
+        make_advertisement("DE:70:E8:B2:39:0C", b"@0\xdd\x03$\x03\x00\x01\x01"),
+    )
+
+    # wait for the event
+    await hass.async_block_till_done()
+
+    dev_reg = async_get_dev_reg(hass)
+    device = dev_reg.async_get_device({get_device_id("DE:70:E8:B2:39:0C")})
+    device_id = device.id
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        CONF_PLATFORM: "device",
+                        CONF_DOMAIN: DOMAIN,
+                        CONF_DEVICE_ID: device_id,
+                        CONF_TYPE: "motion_detected",
+                        CONF_EVENT_PROPERTIES: None,
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {"some": "test_trigger_motion_detected"},
+                    },
+                },
+            ]
+        },
+    )
+
+    message = {
+        CONF_DEVICE_ID: device_id,
+        CONF_ADDRESS: "DE:70:E8:B2:39:0C",
+        EVENT_TYPE: "motion_detected",
+        EVENT_PROPERTIES: None,
+    }
+
+    hass.bus.async_fire(XIAOMI_BLE_EVENT, message)
+    await hass.async_block_till_done()
+
+    assert len(calls) == 1
+    assert calls[0].data["some"] == "test_trigger_motion_detected"
+
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    await hass.async_block_till_done()
+
+
+async def test_validate_trigger_invalid_trigger(hass, caplog):
+    """Test for motion event with invalid triggers."""
+    entry = await _async_setup_xiaomi_motion_device(hass)
+
+    # Emit motion det\ected event so it creates the device in the registry
+    inject_bluetooth_service_info_bleak(
+        hass,
+        make_advertisement("DE:70:E8:B2:39:0C", b"@0\xdd\x03$\x03\x00\x01\x01"),
+    )
+
+    # wait for the event
+    await hass.async_block_till_done()
+
+    dev_reg = async_get_dev_reg(hass)
+    device = dev_reg.async_get_device({get_device_id("DE:70:E8:B2:39:0C")})
+    device_id = device.id
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        CONF_PLATFORM: "device",
+                        CONF_DOMAIN: DOMAIN,
+                        CONF_DEVICE_ID: device_id,
+                        CONF_TYPE: "invalid",
+                        CONF_EVENT_PROPERTIES: None,
+                    },
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {"some": "test_trigger_motion_detected"},
+                    },
+                },
+            ]
+        },
+    )
+
+    assert "value must be one of ['motion_detected']" in caplog.text
```

Comment:
I would make this only check for motion_detected as the string could change in the future

Reply:
like this? 

```python
    assert "motion_detected" in caplog.text
```